### PR TITLE
Restart tasks on dead agents sooner

### PR DESCRIPTION
**Pull request by 6543**
And run the steady queue process as background job

### CODE REVIEW

File: server/queue/fifo.go [hunk 1]
```diff
@@ -51,26 +53,30 @@ type fifo struct {
 	paused        bool
 }
 
+const processTimeInterval = time.Second
+
 // New returns a new fifo queue.
 //
 //nolint:mnd
```
Comment (6543) [inline server/queue/fifo.go:60]:
```suggestion
```

File: server/queue/fifo.go [hunk 2]
```diff
@@ -254,28 +255,36 @@ func (q *fifo) KickAgentWorkers(agentID int64) {
...
 
 // helper function that loops through the queue and attempts to
-// match the item to a single subscriber.
+// match the item to a single subscriber until context got cancel.
 func (q *fifo) process() {
-	q.Lock()
-	defer q.Unlock()
+	for {
+		q.Lock()
+		if q.paused {
```
Comment (lafriks) [inline server/queue/fifo.go:262]: Imho this will use a lot of CPU cycles of paused as it will loop and lock/unlock cpnstantly

Comment (6543) [reply server/queue/fifo.go:262]: ah dammed :/

I had the time wait at the start first but moved it back for the tests

**Comment by 6543** [comment]
backport the queue fix: #4115

**Review by lafriks — APPROVED** [review]

**Pull request merged**